extensions/cactus-plugin-htlc-coordinator-besu/src/main/typescript/web-services/get-openapi-spec.ts

```diff
@@ -0,0 +1,115 @@
+import { Express, Request, Response } from "express";
+
+import {
+  IWebServiceEndpoint,
+  IExpressRequestHandler,
+  IEndpointAuthzOptions,
+} from "@hyperledger/cactus-core-api";
+
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+  IAsyncProvider,
+} from "@hyperledger/cactus-common";
+import {
+  PluginRegistry,
+  registerWebServiceEndpoint,
+} from "@hyperledger/cactus-core";
+import OAS from "../../json/openapi.json";
+
+export interface IGetOpenApiSpecEndpointOptions {
+  logLevel?: LogLevelDesc;
+  pluginRegistry: PluginRegistry;
+}
+
+export class GetOpenApiSpecEndpoint implements IWebServiceEndpoint {
```

Comment:
@charellesandig Please create one endpoint class that is generic across the board and can be re-used by all the plugins when they are registering their web-services.
The `OAS` should be injected through a constructor argument which the plugins have the ability to provide.
This way you don't need to copy paste the same endpoint code to literally every plugin we have so far.

Please also make sure to update the `CLASS_NAME` variable because right now it's set to some other value.

Please also make sure to match the file name to the class name (right now the word endpoint is missing).

Also please add v1 to the classname and the file name as well so classname could be `GetOpenApiSpecV1Endpoint` and the filename `get-open-api-spec-v1-endpoint.ts`

I'm copy pasting here the relevant acceptance criteria that I'm talking about for re-usability:

> The code for this web service should be somehow reused across plugins instead of being copy pasted over to each and every one of them. Probably the cactus-core package is a good place to start, but this might not be the best solution, make sure to check.

Reply:
Working on it Peter. Thank you.